scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala

```diff
@@ -105,25 +137,90 @@ trait H2OMOJOModelUtils {
     json
   }
 
-
-  protected def getModelDetails(mojoData: Array[Byte]): String = {
+  protected def getModelJson(mojoData: Array[Byte]): JsonObject = {
     val is = new ByteArrayInputStream(mojoData)
     val reader = MojoReaderBackendFactory.createReaderBackend(is, MojoReaderBackendFactory.CachingStrategy.MEMORY)
+    ModelJsonReader.parseModelJson(reader)
+  }
 
+  protected def getModelDetails(modelJson: JsonObject): String = {
+    val json = modelJson.get("output").getAsJsonObject
 
-    val modelOutputJson = ModelJsonReader.parseModelJson(reader).getAsJsonObject("output")
-    if (modelOutputJson == null) {
+    if (json == null) {
       "Model details not available!"
     } else {
-      removeMetaField(modelOutputJson)
-      modelOutputJson.remove("domains")
-      modelOutputJson.remove("help")
+      removeMetaField(json)
+      json.remove("domains")
+      json.remove("help")
       val gson = new GsonBuilder().setPrettyPrinting().create
-      val prettyJson = gson.toJson(modelOutputJson)
+      val prettyJson = gson.toJson(json)
       prettyJson
     }
   }
 
+  protected def extractMetrics(json: JsonObject, metricType: String): Map[String, Double] = {
```

Comment:
Has anything changed in the methods that were moved from `H2OModel`?

Reply:
Nope, complete copy